### PR TITLE
feature(ponse) Set Content-Type based on file mime-type.

### PR DESCRIPTION
**Pull request by Daniel-Valentine**
**NOTE.** Not ready for merge into master. Linter complains about the new test case. @coderaiser, could you provide insight into why this is happening?

```
> putout bin lib test .madrun.js
[redacted]/ponse/test/ponse.js
 x:x  error   Cannot read property 'name' of undefine  parser

✖ 1 errors in 1 files
  fixable with the `--fix` option
```

--

# Changes

* `lib/ponse.js`
        - `sendFile()`: Use
        (file-type)[https://www.npmjs.com/package/file-type] (MIT) to
        determine file's mimetype and appropriately set the
        `Content-Type` HTTP header field
* `test/ponse.js`
        - Add test against `Content-Type` field.
        - Files requested in this test are located in
        `test/fixtures/mimetype`. The images are public-domain images
        from Wikimedia Commons:
                - https://commons.wikimedia.org/wiki/File:Floppy_disk_2009_G1.jpg
                - https://commons.wikimedia.org/wiki/File:Railroad1860.png

### CODE REVIEW

File: lib/ponse.js [hunk 1]
```diff
@@ -217,6 +218,11 @@ async function sendFile(params) {
     if (!isGzip)
         p.length = length;
     
+    const [ftError, ftype] = await tryToCatch(filetype.fromFile, p.name);
+    
+    if (!ftError && ftype)
+        p.mime = ftype.mime;
```
Comment (coderaiser) [inline lib/ponse.js:224]:
```suggestion
    const [fileTypeError, fileType] = await tryToCatch(filetype.fromFile, p.name);
    
    if (!fileTypeError && fileType)
        p.mime = fileType.mime;
```

File: test/ponse.js [hunk 2]
```diff
@@ -112,3 +112,28 @@ test('ponse: send', (t) => {
...
 });
 
+test('ponse: get: content-type', async (t) => {
+    const filesToMimes = {
+        "320px-Floppy_disk_2009_G1" : "image/jpeg",
+        "320px-Floppy_disk_2009_G1.jpg" : "image/jpeg",
+        "294px-Railroad1860.png" : "image/png",
+        "294px-Railroad1860" : "image/png",
+        "hello.txt" : "text/plain; charset=UTF-8",
+    }
```
Comment (coderaiser) [inline test/ponse.js:122]: Why some much files :)? I think 3: `jpeg`, `png` and `txt` will be enough :)

File: test/ponse.js [hunk 3]
```diff
@@ -112,3 +112,28 @@ test('ponse: send', (t) => {
...
+
+    var file;
+    for (file in filesToMimes) {
```
Comment (coderaiser) [inline test/ponse.js:125]:
```suggestion
    for (const file of keys(filesToMimes)) {
```

Comment (coderaiser) [reply test/ponse.js:125]: Could you please squash to one commit :)?

File: test/ponse.js (hunk 2 above)
Comment (Daniel-Valentine) [reply test/ponse.js:122]: Part of the motivation for this change was to ensure the `Content-Type` was correctly determined and conveyed for files that didn't have an extension.

I'm fine with deleting the `.{jpg,png}` files and just performing the tests on the extension-less files.

File: test/ponse.js (hunk 3 above)
Comment (Daniel-Valentine) [reply test/ponse.js:125]: There will be a few commits in between to incorporate your changes, but once all is ready I will do a squash.

**Comment by Daniel-Valentine** [comment]
@coderaiser Should I open up the PR in your repo rather than in this fork? It seems like CI is having issues repo: https://travis-ci.com/github/coderaiser/ponse/jobs/324696138#L280

**Comment by Daniel-Valentine** [comment]
@coderaiser: ping

**Comment by coderaiser** [comment]
Could you please rebase and split the loop into separate two-three tests :).

**Comment by coderaiser** [comment]
Merged as https://github.com/coderaiser/node-restafary/commit/4e85f2d67a21795eaaf89091f0075daf1a2c6d73.

**Pull request closed without merging**